test/pg-cdc/mzcompose.py

```diff
@@ -113,6 +116,61 @@ def workflow_replication_disabled(
         c.run_testdrive_files("override/replication-disabled.td")
 
 
+def workflow_silent_connection_drop(
+    c: Composition, parser: WorkflowArgumentParser
+) -> None:
+    pg_version = get_targeted_pg_version(parser)
+    with c.override(
+        create_postgres(
+            pg_version=pg_version,
+            extra_command=[
+                "-c",
+                "wal_sender_timeout=0",
+            ],
+        )
+    ):
+        c.up("materialized", "postgres")
+
+        c.run_testdrive_files(
+            "--no-reset",
+            f"--var=default-replica-size={Materialized.Size.DEFAULT_SIZE}-{Materialized.Size.DEFAULT_SIZE}",
+            "override/silent-connection-drop-part-1.td",
+        )
+
+        _await_postgres_has_no_active_replication_slot(c)
+
+        _claim_postgres_replication_slot(c)
+
+        c.run_testdrive_files("--no-reset", "override/silent-connection-drop-part-2.td")
+
+
+def _await_postgres_has_no_active_replication_slot(c: Composition) -> None:
+    pg_conn = pg8000.connect(
+        host="localhost",
+        user="postgres",
+        password="postgres",
+        port=c.default_port("postgres"),
+    )
+    cursor = pg_conn.cursor()
+
+    for i in range(1, 5):
+        cursor.execute("SELECT active FROM pg_replication_slots;")
+        is_active = cursor.fetchall()[0][0]
+
+        if is_active:
+            time.sleep(1)
+            continue
+        else:
+            return
+
+    raise RuntimeError("Replication slot is still active")
+
+
+def _claim_postgres_replication_slot(c: Composition) -> None:
+    # TODO
```

Comment:
@benesch, do you have a suggestion on how to claim the replication slot?
I know how it could be done with psycopg2 but we are using psycopg3, which does no longer support this way (creating a connection with `psycopg2.extras.LogicalReplicationConnection` as connection factory and calling `start_replication` on the cursor)...

Reply:
Yeah, looks like psycopg3 doesn't support this. Too bad! Easiest thing to do IMO would be to use the `pg_recvlogical` command in the `postgres` Docker image. Here's a very quick example:

```
docker run postgres:latest pg_recvlogical --start --slot materialize_49265401a63e4c37b287857d9ac54997 --file=- --dbname=DBNAME -o proto_version=1 -o publication_names=pubname
```

Would need to be adapted to point at the running PostgreSQL server and use the right database/auth etc. But should be pretty easy to rig up!

Reply:
Thank you!

Reply:
Using this command through mzcompose in https://github.com/MaterializeInc/materialize/pull/27727/commits/81d7d2091bc0d492404259c4c7f5da8a4da9dc56, I am now running into:
```
pg_recvlogical: error: could not send replication command "START_REPLICATION SLOT "materialize_f24fb996417f410faf66884f7bffe692" LOGICAL 0/0": ERROR:  client sent proto_version=0 but server only supports protocol 1 or higher
CONTEXT:  slot "materialize_f24fb996417f410faf66884f7bffe692", output plugin "pgoutput", in the startup callback
```

Some answers on Stackoverflow say that output plugin should be changed to `wal2json` but the plugin option description is:

> When creating a slot, use the specified logical decoding output plugin. See [Chapter 49](https://www.postgresql.org/docs/current/logicaldecoding.html). This option has no effect if the slot already exists.

@benesch, any ideas?

Reply:
Make sure to include the `-o` flags like in the `docker run` example I shared!

```
-o proto_version=1 -o publication_names=pubname
```

You’ll need to adjust `pubname` appropriately.

Reply:
Oh, yes, you are right! Perfect, it works now! 🎉